src/test/java/tests/SingDuetSongTest.java

```diff
@@ -0,0 +1,111 @@
+package tests;
+
+import Page_Objects.TestBase;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+
+public class SingDuetSongTest {
+
+    private ChromeDriver driver;
+    private TestBase testBase;
+    private Actions actions;
+
+    String songID = "mana-y-prince-royce-el-verdadero-amor-perdona";
+    String songLanguage = "Spanish";
+    String playlistName = "Duets";
+    String groupName = "M";
+    String difficultyLevel = "Hard";
+    String gameMode = "Duel";
+    String player1Name = "all";
+    String player2Name = "Karaoke";
+    String updatedName = "I won - heheheh";
+    int player1 = 0;
+    int player2 = 1;
+
+
+    @BeforeEach
+    public void setUp() {
+        driver = new ChromeDriver();
+        testBase = new TestBase(driver);
+        actions = new Actions(driver);
+        testBase.setUp();
+    }
+
+    @Test
+    public void singingDuetSong() {
+        testBase.getLandingPage().enterTheGame();
+
+        // Step 1: Select Advanced setup
+        testBase.getInputSelectionPage().selectAdvancedSetup();
+        testBase.getAdvancedConnectionPage().saveAndGoToSing();
+
+        // Step 2: Ensure song language is selected
+        testBase.getMainMenuPage().goToSongList();
+        testBase.getSongLanguagesPage().ensureSongLanguageIsSelected(songLanguage);
+        testBase.getSongLanguagesPage().continueAndGoToSongList();
+
+        // Step 3: Go to playlist and open preview the song
+        testBase.getFullscreenPage().isFullscreenOn();
+        testBase.getSongListPage().goToPlaylist(playlistName);
+        testBase.getSongListPage().expectPlaylistToBeSelected(playlistName);
+        testBase.getSongListPage().goToGroupName(groupName);

```

Comment:
is it needed? (another test checks that)

Reply:
You're right - I removed the line with `goToGroupName(groupName)` because that method is already present in ScrollToLetterTest, but `expectPlaylistToBeSelected(playlistName)` is a new method, only used in this test, so can I keep it?